### DISCUSSION TITLE
Hosting multiple PB instances on a single server.

### DISCUSSION THREAD

**Discussion by lia-07** [pocketbase/pocketbase#1734] [Q&A] [answered]
Hi, I aim to have multiple PB instances running on a single Ubuntu server, behind an Nginx reverse proxy. I managed to get one instance working perfectly through this reverse proxy, but ran into issues when I tried to have more than one, and gave them their own route.

For context, in the root directory I have a `pb` folder, which contains the `pocketbase` executable, and now two seperate data directories, `example` and `test`. I followed the steps in [this discussion](https://github.com/pocketbase/pocketbase/discussions/738), and ran the commands `./pocketbase serve --dir="./example" --http="127.0.0.1:8091"` and `./pocketbase serve --dir="./test" --http="127.0.0.1:8092"`. 

Then I attempted to reverse proxy both of them behind seperate paths. When I went to `123.45.678.9/example/` in my browser, it redirects to `123.45.678.9/_/?installer#` as if the instance isn't supposed to be behind a different path, and then Nginx serves me a 404 page. It does the same when attempting to navigate to `123.45.678.9/test/`.

When I specifically navigate to `http://123.45.678.9/example/?installer#`, it shows the expected page, although the form doesn't work. 

I am new to Nginx and PocketBase, so I apologise if this is a bad question.

Any help would be appreciated.

Here is my http block in my Nginx config:
```
http {
    server {
        listen 80;
        server_name 123.45.678.9;

        location /example/ {
            proxy_pass http://127.0.0.1:8091/_/;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }

        location /example/api/ { 
            proxy_pass http://127.0.0.1:8091/_/api/;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }

        location /test/ {
            proxy_pass http://127.0.0.1:8092/_/;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }

        location /test/api/ {
            proxy_pass http://127.0.0.1:8092/api/;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }
    }
```

**Comment by Vicquemare** [comment]
If you're using the compiled binary, from what I've seen the Admin UI expects the backend server to be started at http://localhost:8090 where localhost would be the same as 127.0.0.1, but you can change that by creating a new env file with PB_BACKEND_URL = your_address variable inside it, for that I think you would have to compile pocketbase yourself  or [use it as Framework](https://pocketbase.io/docs/use-as-framework/), depending on your use case tho, I would recomend using docker for both instances and using something like nginx reverse proxy with docker, maybe both in different subdomains if you're gonna be using a domain, not sure if this would be ideal for your use case but its just an idea, let me know if it helps!

**Reply by dschissler** [reply]
@Vicquemare The Admin UI runs on the same host setting as the API.  So if you change `http` then the admin also changes.

**Comment by ganigeorgiev** [comment]
@lia-07 When proxying a non-root path you may have to disable or configure the [`proxy_redirect`](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_redirect) directive. Also you usually shouldn't need 2 separate proxies for the admin and api endpoints. I haven't tested it, but you can try something like:

```nginx
location /example/ {
    # check http://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive
    proxy_set_header Connection '';
    proxy_http_version 1.1;
    proxy_read_timeout 360s;

    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;

    # disable proxy redirect or configure it per
    # https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_redirect
    proxy_redirect off;

    proxy_pass http://127.0.0.1:8091;
}

# you may need a non-trailing slash redirect
# or depending on your use case you can also change the first location pattern to consider it
location /example {
    return 301 /example/;
}
```

**Comment by izulwahidin** [comment]
Check this out [PocketBase Multi Instance](https://github.com/izulwahidin/Docker-PocketBase-Multi-Instance)

**Comment by TobiasNickel** [comment]
I wonder if you need to rewrite the path, because pocketbase expose the adminportal at `/_/`.

personally I use routing based on subdomain. that works perfectly. (I use caddy as my reverse proxy)

**Comment by ganigeorgiev** [comment, marked as answer]
_I'm closing this as it is an old discussion. If someone else has a similar problem please open a new discussion._

In any case, for example reverse proxy config please refer to the ["Using reverse proxy"](https://pocketbase.io/docs/going-to-production/#using-reverse-proxy).
